ihttp/ihttp.go

```diff
@@ -97,6 +99,18 @@ func (s Handler) Handle(w http.ResponseWriter, req *http.Request) {
 
 		return
 	}
+
+	if len(s.Patch) > len(binary.MagicString) {
+		log.Info("patch string is too long, ignoring")
+	} else if len(s.Patch) > 0 {
+		index := bytes.Index(file, binary.MagicString)
+		if index >= 0 {
+			copy(file[index:], s.Patch)
```

Comment:
Separate to above, and as I mentioned before, I think this would read a little cleaner if we first check if a patch needs to be applied:

```go
if len(s.Patch) > 0 {
  if len(s.Patch) > len(binary.MagicString) {
    // Return 400 response because the user supplied invalid input
  }

  // Patch binary
}

Reply:
We might also consider moving the logic to the `binary` package as a patch func.

```go
// /binary/patch.go

var magicString = []byte(...)

func Patch(content, patch []byte) ([]byte, error) {
	if len(patch) == 0 {
		return content, nil
	}

	if len(patch) > len(magicString) {
		return nil, ErrInvalidPatch
	}
	
	i := bytes.Index(content, magicString)

	// If there's no magic string the content doesn't support patching.
	if i == -1 {
		return content, nil
	}
	
	// Duplicate the content before applying the patch so we don't overwrite
	// the underlying array.
	dup := make([]byte, len(content))
	copy(dup, content)
	copy(dup[i:], patch)

	return dup, nil
}
```

@jacobweinstock